testbed/go.mod

```diff
@@ -7,7 +7,7 @@ require (
 	github.com/census-instrumentation/opencensus-proto v0.2.1
 	github.com/go-ole/go-ole v1.2.4 // indirect
 	github.com/golang/protobuf v1.3.2
-	github.com/open-telemetry/opentelemetry-collector v0.2.0-0.20200110151913-9777072f35b3
+	github.com/open-telemetry/opentelemetry-collector v0.2.4-0.20200115225140-264426a9cae4
```

Comment:
v0.2.4 does not exist. How does this work?

Reply:
Using `go get github.com/open-telemetry/opentelemetry-collector@HEAD` automatically adds `<latest_released_version + 1>-<timestamp>-<hash>`. It treats commits newer than latest release as a pre-release of the next version in line. So `v0.2.4-0-<timestamp>-<hash>` suggests that this version will eventually become `v0.2.4` when it is released or in other words that it is a development/pre-alpha release of `v0.2.4`. For better or worse that is the convention adopted by the tool.

Reply:
Also when commit hash is provided, go always uses it to resolve the exact copy of dependency. It doesn't use the specified version to resolve the source code in this case but it might validate the version number so it makes sense.